lib/string_reverse.rb

```diff
@@ -1,6 +1,19 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is the number of iterations in the while loop
+# Space complexity: Constant or O(1), because the additional stirage needed is just the variable "a"
 def string_reverse(my_string)
-  raise NotImplementedError
+  if my_string == nil
+    return nil
+  end
+  i = 0
+  j = my_string.length
+
+  while i < j
+    a = my_string[i]
+    my_string[i] = my_string[j - 1]
+    my_string[j - 1] = a

```

Comment:
If you wanted, you could do this on one line:

```ruby
my_string[i], my_string[j] = my_string[j], my_string[i]
```

but it's just as valid to use a temporary variable.